lua/starfall/libs_cl/midi.lua

```diff
@@ -0,0 +1,82 @@
+if not SF.Require("midi") then return function() end end
+
+--Midi Library
+-- @name midi
+-- @class library
+-- @libtbl midi_library
+SF.RegisterLibrary("midi")
+
+return function(instance)
+-- Only usable by the owner of the starfall chip
+if LocalPlayer() ~= instance.player then return end
+
+local midi_library = instance.Libraries.midi
+
+-- Event hook for midi
+-- @class hook
+-- @client
+-- @libtbl midi_library
+-- @param number time the exact systime which the event occured
+-- @param number command the command code of the event.  First 4 bits are the command code and last 4 are the channel
+-- @param varargs the different parameters of the midi event.  Can have 1 or more.
+SF.hookAdd("MIDI", "midi")
+
+-- Gets a table of all midi devices' ports.
```

Comment:
These should just be midi_library.x = midi.y since they don't need to be sandboxed (And we don't sandbox binary module libs anyway) ((See xinput))

Reply:
kept some of them protected behind checkluatype